packages/protocol/contracts/shared/tokenvault/ERC20Vault.sol

```diff
@@ -404,11 +445,32 @@ contract ERC20Vault is BaseVault {
             IERC20 t = IERC20(_op.token);
             uint256 _balance = t.balanceOf(address(this));
             t.safeTransferFrom(msg.sender, address(this), _op.amount);
-            balanceChange_ = t.balanceOf(address(this)) - _balance;
+            balanceChangeAmount_ = t.balanceOf(address(this)) - _balance;
+
+            if (_op.solverFee > 0) {
+                _balance = t.balanceOf(address(this));
+                t.safeTransferFrom(msg.sender, address(this), _op.solverFee);
+                balanceChangeSolverFee_ = t.balanceOf(address(this)) - _balance;
+            }
+        }
+
+        // Prepare solver condition for allowing fast withdrawal on L1
+        if (_op.solverFee > 0) {
```

Comment:
```suggestion
        if (_op.solverFee != 0) {
```

Reply:
Clarification requested in a similar suggested change below